### PR TITLE
[wip] Send slack message if self-scheduled runner fails

### PR DESCRIPTION
**Pull request by sshleifer**

### CODE REVIEW

**Comment by sshleifer** [comment]
@julien-c do you know an easy way to test whether this works without merging?

File: .github/workflows/self-scheduled.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
 name: Self-hosted runner (scheduled)
-
+env:
+  TF_FORCE_GPU_ALLOW_GROWTH: "true"
+  OMP_NUM_THREADS: 1
+  RUN_SLOW: yes
+  USE_CUDA: yes
```
Comment (julien-c) [inline .github/workflows/self-scheduled.yml:6]: those envs are only for one command so I'm fine with where they were

File: .github/workflows/self-scheduled.yml [hunk 2]
```diff
@@ -12,39 +18,41 @@ jobs:
...
-        python -m venv .env
-        source .env/bin/activate
-        which python
-        python --version
-        pip --version
-    - name: Install dependencies
-      run: |
-        source .env/bin/activate
-        pip install .[sklearn,torch,testing]
+      - uses: actions/checkout@v2
```
Comment (julien-c) [inline .github/workflows/self-scheduled.yml:21]: please no whitespace-only change

File: .github/workflows/self-scheduled.yml [hunk 3]
```diff
@@ -12,39 +18,41 @@ jobs:
...
-        USE_CUDA: yes
-      run: |
-        source .env/bin/activate
-        python -m pytest -n 1 --dist=loadfile -s -v ./tests/
-        
+      - name: Run all tests on GPU
+        run: |
+          source .env/bin/activate
+          python -m pytest -n 1 --dist=loadfile -s -v ./tests/
+      - if: failure()
```
Comment (julien-c) [inline .github/workflows/self-scheduled.yml:51]: I'm fine with this though there are other options that don't require modifying this file

**Review by julien-c — CHANGES_REQUESTED** [review]

**Pull request closed without merging**